app_dart/lib/src/request_handlers/scheduler/batch_backfiller.dart

```diff
@@ -80,7 +80,25 @@ class BatchBackfiller extends RequestHandler {
     log.fine('Backfilling ${backfill.length} builds');
     log.fine(backfill.map<String>((Tuple<Target, FullTask, int> tuple) => tuple.first.value.name));
 
-    // Create list of backfill requests.
+    // Update tasks status as in progress to avoid duplicate scheduling.
+    final List<Task> backfillTasks = backfill.map((Tuple<Target, FullTask, int> tuple) => tuple.second.task).toList();
+    try {
+      await datastore.withTransaction<void>((Transaction transaction) async {
+        transaction.queueMutations(inserts: backfillTasks);
+        await transaction.commit();
+        log.fine(
+            'Updated ${backfillTasks.length} tasks: ${backfillTasks.map((e) => e.name).toList()} when backfilling.');
+      });
+      // Schedule all builds asynchronously.
+      // Schedule after db updates to avoid duplicate scheduling when db update fails.
+      await Future.wait<void>(backfillRequestList(backfill));
+    } catch (error) {
+      log.severe('Failed to update tasks when backfilling: $error');
+    }
+  }
+
+  // Create list of backfill requests.
```

Comment:
nit: should this be '///' instead since it is a doc string?

Reply:
Yeah, you are right (copied form earlier comment). Updated.